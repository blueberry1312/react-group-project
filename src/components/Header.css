header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 70px;
  margin: 0 100px 0 100px;
}

nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  padding: 0 20px;
}

.logo-container {
  display: flex;
  align-items: center;
}

.logo {
  margin-right: 20px;
}

.logo img {
  max-height: 50px;
}

.title {
  font-size: 1.2rem;
  font-weight: bold;
}

ul {
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
}

li {
  margin-left: 20px;
}

a {
  text-decoration: none;
  color: #333;
  transition: all 0.2s ease-in-out;
}

a:hover {
  color: #007bff;
}

.active-link {
  color: #007bff;
}
