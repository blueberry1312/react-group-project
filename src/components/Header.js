import React from 'react';
import { NavLink } from 'react-router-dom';
import './Header.css';
import Logo from '../assets/planet-1.png';

const Header = () => (
  <header>
    <nav>
      <div className="logo-container">
        <div className="logo">
          <img src={Logo} alt="Logo" />
        </div>
        <div className="title">
          <h1>Space Travelers Hub</h1>
        </div>
      </div>
      <ul>
        <li><NavLink to="/rockets" activeClassName="active-link">Rockets</NavLink></li>
        <li><NavLink to="/missions" activeClassName="active-link">Missions</NavLink></li>
        <li><NavLink to="/my-profile" activeClassName="active-link">My Profile</NavLink></li>
      </ul>
    </nav>
  </header>
);

export default Header;
