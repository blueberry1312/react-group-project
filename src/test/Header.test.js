import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from '../components/Header';

describe('Header test', () => {
  test('Header component is displayed on the HTML', () => {
    render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );
    const header = screen.getByTitle('header-component');
    expect(header).toMatchSnapshot();
  });
});
