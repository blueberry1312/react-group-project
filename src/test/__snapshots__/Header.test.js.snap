// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test Header component is displayed on the HTML 1`] = `
<header
  title="header-component"
>
  <nav>
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      <div
        class="logo-container"
      >
        <div
          class="logo"
        >
          <img
            alt="Logo"
            src="planet-1.png"
          />
        </div>
        <div
          class="title"
        >
          <h1>
            Space Travelers Hub
          </h1>
        </div>
      </div>
    </a>
    <ul>
      <li>
        <a
          activeclassname="active-link"
          class=""
          href="/rockets"
          style="text-decoration: none;"
        >
          Rockets
        </a>
      </li>
      <div
        class="vert-lane"
      />
      <li>
        <a
          activeclassname="active-link"
          class=""
          href="/missions"
          style="text-decoration: none;"
        >
          Missions
        </a>
      </li>
      <div
        class="vert-lane"
      />
      <li>
        <a
          activeclassname="active-link"
          class=""
          href="/my-profile"
          style="text-decoration: none;"
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
